src/Articulate/Controllers/ArticulateSearchController.cs

```diff
@@ -77,27 +77,8 @@ public ActionResult Search(RenderModel model, string term, string provider = nul
 
             int totalPosts;
             var searchResult = ArticulateSearcher.Search(term, provider, rootPageModel.BlogArchiveNode.Id, rootPageModel.PageSize, p.Value - 1, out totalPosts);
-            
-            var pageSize = rootPageModel.PageSize;
 
-            var totalPages = totalPosts == 0 ? 1 : Convert.ToInt32(Math.Ceiling((double)totalPosts / pageSize));
-
-            //Invalid page, redirect without pages
-            if (totalPages < p)
-            {
-                return new RedirectToUmbracoPageResult(model.Content.Parent, UmbracoContext);
-            }
-
-            var pager = new PagerModel(
-                pageSize,
-                p.Value - 1,
-                totalPages,
-                totalPages > p ? model.Content.Url.EnsureEndsWith('?') + "term=" + term + "&p=" + (p + 1) : null,
-                p > 2 ? model.Content.Url.EnsureEndsWith('?') + "term=" + term + "&p=" + (p - 1) : p > 1 ? model.Content.Url.EnsureEndsWith('?') + "term=" + term : null);
-
-            var listModel = new ListModel(tagPage, searchResult, pager);
-
-            return View(PathHelper.GetThemeViewPath(listModel, "List"), listModel);
+            return GetPagedListView(model, searchPage, searchResult, searchResult.Count(), p);
```

Comment:
searchResult.Count() is not the total item count, that is only the count of the enumerated search results (which is inefficient), the total search results is from the variable `totalPosts`